File: 230-二叉搜索树中第k小的元素/230_1.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def kthSmallest(self, root, k):
        """
        :type root: TreeNode
        :type k: int
        :rtype: int
        """
        self.res = []
        def inorder(root):
            if not root:
                return
            inorder(root.left)
            self.res.append(root.val)
            print(root.val)
            inorder(root.right)
        inorder(root)
        sorted(self.res)
        return self.res[k-1]

File: 230-二叉搜索树中第k小的元素/230_1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
  public int index = 0;
  public int res = 0;
  public int kthSmallest(TreeNode root, int k) {
      if(root == null) return 0;
      index = k;
      dfs(root);
      return res;
  }
  public void dfs(TreeNode node){
      if(node == null || index < 0){
          return;
      }
      dfs(node.left);
      index--;
      if(index == 0){
          res = node.val;
      }
      dfs(node.right);
  }
}